src/main/java/teammates/ui/controller/Action.java

```diff
@@ -576,10 +584,131 @@ private boolean isMasqueradeModeRequested(AccountAttributes loggedInUser, String
                && loggedInUser.googleId != null
                && !loggedInUser.googleId.equals(requestedUserId);
     }
+
+    /**
+     *Common method to Get the feedback data
+     */
+
+    protected FeedbackSessionAttributes extractFeedbackSessionData() {
+
+        FeedbackSessionAttributes newSession = new FeedbackSessionAttributes();
+
+        newSession.setCourseId(getRequestParamValue(Const.ParamsNames.COURSE_ID));
+        newSession.setFeedbackSessionName(Sanitizer.sanitizeTitle(
+                getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME)));
+
+        // getting the class name from which it's being called
+        String className = new Exception().getStackTrace()[1].getClassName();
+
+        if ("InstructorFeedbackEditSaveAction".equals(className)) {
```

Comment:
@AnaghHegde 
How about splitting `extractFeedbackSessionData` into several smaller methods? Then perhaps you will have one small method whose actions need to differ based on the class they are called on, which can be easily overriden.

In particular,
```
if ("InstructorFeedbackEditSaveAction".equals(className)) {
   newSession.setCreatorEmail(getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_CREATOR));
} else {
  newSession.setCreatedTime(new Date());
}
```
can be extracted into a smaller method, which does `newSession.setCreatedTime(new Date())` in the base class and can be overriden to do `newSession.setCreatorEmail(getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_CREATOR))` inside `InstructorFeedbackEditSaveAction`. This method can then be called from `extractFeedbackSessionData`, and binds to either the base class version or the subclass version depending on the type of the object from which it is being called.